File: Project/TeamA_Enterprise-Learning-Platform-main (1)/TeamA_Enterprise-Learning-Platform-main/skillsphere-backend/src/main/java/com/skillsphere/service/AuthService.java
package com.skillsphere.service;

import com.skillsphere.dto.LoginRequest;
import com.skillsphere.dto.LoginResponse;
import com.skillsphere.dto.RegisterRequest;

public interface AuthService {

    LoginResponse register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    void forgotPassword(String email);

    boolean validateResetToken(String token);

    void resetPassword(String token, String newPassword);
}

File: Project/TeamA_Enterprise-Learning-Platform-main (1)/TeamA_Enterprise-Learning-Platform-main/skillsphere-frontend/src/services/AuthService.js
import apiClient from '../api/apiClient';
import { API_ENDPOINTS } from '../constants/api';

const AuthService = {
  /**
   * Login with email and password
   * @returns {Promise} { token, email, role }
   */
  login(email, password, selectedRole, rememberMe = false) {
    return apiClient.post(API_ENDPOINTS.LOGIN, { email, password, selectedRole, rememberMe });
  },

  /**
   * Register a new student account
   */
  register(payload) {
    return apiClient.post(API_ENDPOINTS.REGISTER, payload);
  },

  forgotPassword(email) {
    return apiClient.post('/api/auth/forgot-password', { email });
  },

  validateResetToken(token) {
    return apiClient.get(`/api/auth/validate-reset-token?token=${encodeURIComponent(token)}`);
  },

  resetPassword(token, newPassword) {
    return apiClient.post('/api/auth/reset-password', { token, newPassword });
  }
};

export default AuthService;
